src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,61 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int DEFAULT_LENGTH = 10;
+    private int size;
+    private StorageObject<K, V>[] storage;
+
+    public StorageImpl() {
+        storage = new StorageObject[DEFAULT_LENGTH];
+    }
+
+    private class StorageObject<K, V> {
+        private K key;
+        private V value;
+
+        public StorageObject(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = getIndexByKey(key);
+        if (index != -1) {
```

Comment:
```suggestion
        if (index != NO_INDEX) {
```
let's create constant for magic number